// ngx-slickjs/esm5/lib/ngx-slickjs.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from "@angular/core";
import { SlickContainerDirective } from "./directives/slick-container.directive";
import { SlickItemDirective } from "./directives/slick-item.directive";
import { SLICK_LINKS, linkOptionsFactory } from "./tokens/links.token";
import * as ɵngcc0 from '@angular/core';
var NgxSlickJsModule = /** @class */ (function () {
    function NgxSlickJsModule() {
    }
    /**
     * @param {?=} __0
     * @return {?}
     */
    NgxSlickJsModule.forRoot = /**
     * @param {?=} __0
     * @return {?}
     */
    function (_a) {
        var _b = (_a === void 0 ? (/** @type {?} */ ({})) : _a).links, links = _b === void 0 ? {} : _b;
        return {
            ngModule: NgxSlickJsModule,
            providers: [
                {
                    provide: SLICK_LINKS,
                    useValue: links,
                },
                {
                    provide: "slick-links",
                    useFactory: linkOptionsFactory,
                    deps: [SLICK_LINKS],
                },
            ],
        };
    };
NgxSlickJsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxSlickJsModule });
NgxSlickJsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxSlickJsModule_Factory(t) { return new (t || NgxSlickJsModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxSlickJsModule, { declarations: function () { return [SlickContainerDirective, SlickItemDirective]; }, exports: function () { return [SlickContainerDirective, SlickItemDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxSlickJsModule, [{
        type: NgModule,
        args: [{
                declarations: [SlickContainerDirective, SlickItemDirective],
                exports: [SlickContainerDirective, SlickItemDirective]
            }]
    }], function () { return []; }, null); })();
    return NgxSlickJsModule;
}());
export { NgxSlickJsModule };
