// ngx-slickjs/esm2015/lib/ngx-slickjs.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from "@angular/core";
import { SlickContainerDirective } from "./directives/slick-container.directive";
import { SlickItemDirective } from "./directives/slick-item.directive";
import { SLICK_LINKS, linkOptionsFactory } from "./tokens/links.token";
import * as ɵngcc0 from '@angular/core';
export class NgxSlickJsModule {
    /**
     * @param {?=} __0
     * @return {?}
     */
    static forRoot({ links = {} } = (/** @type {?} */ ({}))) {
        return {
            ngModule: NgxSlickJsModule,
            providers: [
                {
                    provide: SLICK_LINKS,
                    useValue: links,
                },
                {
                    provide: "slick-links",
                    useFactory: linkOptionsFactory,
                    deps: [SLICK_LINKS],
                },
            ],
        };
    }
}
NgxSlickJsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxSlickJsModule });
NgxSlickJsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxSlickJsModule_Factory(t) { return new (t || NgxSlickJsModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxSlickJsModule, { declarations: function () { return [SlickContainerDirective, SlickItemDirective]; }, exports: function () { return [SlickContainerDirective, SlickItemDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxSlickJsModule, [{
        type: NgModule,
        args: [{
                declarations: [SlickContainerDirective, SlickItemDirective],
                exports: [SlickContainerDirective, SlickItemDirective]
            }]
    }], null, null); })();
