// ngx-sharebuttons/esm2015/lib/share.module.js
import { NgModule } from '@angular/core';
import { PlatformModule } from '@angular/cdk/platform';
import { ClipboardModule } from '@angular/cdk/clipboard';
import { ShareDirective } from './share-button.directive';
import { SHARE_BUTTONS_CONFIG } from './share.models';
import * as ɵngcc0 from '@angular/core';
export class ShareModule {
    static withConfig(config) {
        return {
            ngModule: ShareModule,
            providers: [
                { provide: SHARE_BUTTONS_CONFIG, useValue: config }
            ]
        };
    }
}
ShareModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ShareModule });
ShareModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ShareModule_Factory(t) { return new (t || ShareModule)(); }, imports: [[
            PlatformModule,
            ClipboardModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ShareModule, { declarations: function () { return [ShareDirective]; }, imports: function () { return [PlatformModule,
        ClipboardModule]; }, exports: function () { return [ShareDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ShareModule, [{
        type: NgModule,
        args: [{
                imports: [
                    PlatformModule,
                    ClipboardModule
                ],
                declarations: [
                    ShareDirective
                ],
                exports: [
                    ShareDirective
                ]
            }]
    }], null, null); })();
