// ngx-slickjs/esm2015/lib/directives/slick-item.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, ElementRef, Host } from "@angular/core";
import { SlickContainerDirective } from "./slick-container.directive";
import * as ɵngcc0 from '@angular/core';
export class SlickItemDirective {
    /**
     * @param {?} elRef
     * @param {?} slickRef
     */
    constructor(elRef, slickRef) {
        this.elRef = elRef;
        this.slickRef = slickRef;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        this.slickRef.addSlide(this.elRef.nativeElement);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.slickRef.removeSlide(this.elRef.nativeElement);
    }
}
SlickItemDirective.ɵfac = function SlickItemDirective_Factory(t) { return new (t || SlickItemDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(SlickContainerDirective, 1)); };
SlickItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SlickItemDirective, selectors: [["", "slickItem", ""]] });
/** @nocollapse */
SlickItemDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: SlickContainerDirective, decorators: [{ type: Host }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlickItemDirective, [{
        type: Directive,
        args: [{
                selector: "[slickItem]"
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: SlickContainerDirective, decorators: [{
                type: Host
            }] }]; }, null); })();
if (false) {
    /** @type {?} */
    SlickItemDirective.prototype.elRef;
    /**
     * @type {?}
     * @private
     */
    SlickItemDirective.prototype.slickRef;
}
