// paytm-blink-checkout-angular/esm5/lib/checkout.module.js
import { NgModule } from '@angular/core';
import { CheckoutComponent } from './components/checkout.component';
import { CommonModule } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
var CheckoutModule = /** @class */ (function () {
    function CheckoutModule() {
    }
CheckoutModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CheckoutModule });
CheckoutModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CheckoutModule_Factory(t) { return new (t || CheckoutModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CheckoutModule, { declarations: function () { return [CheckoutComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CheckoutComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CheckoutModule, [{
        type: NgModule,
        args: [{
                declarations: [CheckoutComponent],
                imports: [CommonModule],
                exports: [CheckoutComponent]
            }]
    }], function () { return []; }, null); })();
    return CheckoutModule;
}());
export { CheckoutModule };
