// ngx-sharebuttons/esm2015/popup/share-popup-buttons.directive.js
import { Directive, EventEmitter, HostListener, Input, Output } from '@angular/core';
import { SharePopupService } from './share-popup.service';
// import { ShareService, SHARE_BUTTONS } from '../../src/public-api';
import { ShareService, SHARE_BUTTONS } from 'ngx-sharebuttons';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'ngx-sharebuttons';
export class SharePopupButtonsDirective {
    constructor(_sharePopup, _share) {
        this._sharePopup = _sharePopup;
        this._share = _share;
        this.theme = this._share.config.theme;
        this.include = Object.keys(SHARE_BUTTONS);
        this.closeIcon = ['fas', 'times-circle'];
        /** Show buttons icons */
        this.showIcon = true;
        /** Show buttons text */
        this.showText = false;
        /** Buttons size */
        this.size = 0;
        this.hasBackdrop = true;
        /** Share dialog opened event */
        this.opened = new EventEmitter();
        /** Share dialog closed event */
        this.closed = new EventEmitter();
        // Stream that emits when the share popup is closed
        this.popupClosed = new EventEmitter();
        this._popupClosed = this.popupClosed.subscribe(() => this.close());
    }
    onClick(e) {
        this._sharePopup.open({
            theme: this.theme,
            include: this.include,
            exclude: this.exclude,
            url: this.url,
            title: this.title,
            description: this.description,
            image: this.image,
            tags: this.tags,
            autoSetMeta: this.autoSetMeta,
            show: this.show,
            showIcon: this.showIcon,
            showText: this.showText,
            size: this.size,
            disabled: this.disabled,
            opened: this.opened,
            closed: this.closed,
            closeIcon: this.closeIcon,
            closeClick: this.popupClosed,
            hasBackdrop: this.hasBackdrop,
            backdropClass: this.backdropClass
        });
    }
    ngOnDestroy() {
        this._popupClosed.unsubscribe();
    }
    close() {
        this._sharePopup.close();
    }
}
SharePopupButtonsDirective.ɵfac = function SharePopupButtonsDirective_Factory(t) { return new (t || SharePopupButtonsDirective)(ɵngcc0.ɵɵdirectiveInject(SharePopupService), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ShareService)); };
SharePopupButtonsDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SharePopupButtonsDirective, selectors: [["", "shareButtonsPopup", ""]], hostBindings: function SharePopupButtonsDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function SharePopupButtonsDirective_click_HostBindingHandler($event) { return ctx.onClick($event); });
    } }, inputs: { theme: "theme", include: "include", closeIcon: "closeIcon", showIcon: "showIcon", showText: "showText", size: "size", hasBackdrop: "hasBackdrop", exclude: "exclude", url: "url", title: "title", description: "description", image: "image", tags: "tags", autoSetMeta: "autoSetMeta", show: "show", disabled: "disabled", backdropClass: "backdropClass" }, outputs: { opened: "opened", closed: "closed", popupClosed: "popupClosed" } });
SharePopupButtonsDirective.ctorParameters = () => [
    { type: SharePopupService },
    { type: ShareService }
];
SharePopupButtonsDirective.propDecorators = {
    theme: [{ type: Input }],
    include: [{ type: Input }],
    exclude: [{ type: Input }],
    closeIcon: [{ type: Input }],
    url: [{ type: Input }],
    title: [{ type: Input }],
    description: [{ type: Input }],
    image: [{ type: Input }],
    tags: [{ type: Input }],
    autoSetMeta: [{ type: Input }],
    show: [{ type: Input }],
    showIcon: [{ type: Input }],
    showText: [{ type: Input }],
    size: [{ type: Input }],
    disabled: [{ type: Input }],
    hasBackdrop: [{ type: Input }],
    backdropClass: [{ type: Input }],
    opened: [{ type: Output }],
    closed: [{ type: Output }],
    popupClosed: [{ type: Output }],
    onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SharePopupButtonsDirective, [{
        type: Directive,
        args: [{
                selector: '[shareButtonsPopup]'
            }]
    }], function () { return [{ type: SharePopupService }, { type: ɵngcc1.ShareService }]; }, { theme: [{
            type: Input
        }], include: [{
            type: Input
        }], closeIcon: [{
            type: Input
        }], showIcon: [{
            type: Input
        }], showText: [{
            type: Input
        }], size: [{
            type: Input
        }], hasBackdrop: [{
            type: Input
        }], opened: [{
            type: Output
        }], closed: [{
            type: Output
        }], popupClosed: [{
            type: Output
        }], onClick: [{
            type: HostListener,
            args: ['click', ['$event']]
        }], exclude: [{
            type: Input
        }], url: [{
            type: Input
        }], title: [{
            type: Input
        }], description: [{
            type: Input
        }], image: [{
            type: Input
        }], tags: [{
            type: Input
        }], autoSetMeta: [{
            type: Input
        }], show: [{
            type: Input
        }], disabled: [{
            type: Input
        }], backdropClass: [{
            type: Input
        }] }); })();
