// ngx-sharebuttons/esm2015/popup/share-popup.component.js
import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
import { SharePopupService } from './share-popup.service';
// import { ShareService } from '../../src/public-api';
import { ShareService } from 'ngx-sharebuttons';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'ngx-sharebuttons';
import * as ɵngcc2 from './share-popup-buttons.directive';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from '@fortawesome/angular-fontawesome';

function SharePopupComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵelement(1, "fa-icon", 5);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("icon", ctx_r0.icon)("fixedWidth", true);
} }
function SharePopupComponent_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r1.text, " ");
} }
export class SharePopupComponent {
    constructor(_sharePopup, _share) {
        this._sharePopup = _sharePopup;
        this._share = _share;
        this.theme = this._share.config.theme;
        this.icon = ['fas', 'share'];
        this.text = 'Share';
        this.closeIcon = ['fas', 'times-circle'];
        /** Show buttons icons */
        this.showIcon = true;
        /** Show buttons text */
        this.showText = false;
        /** Buttons size */
        this.size = 0;
        this.hasBackdrop = true;
        /** Share dialog opened event */
        this.opened = new EventEmitter();
        /** Share dialog closed event */
        this.closed = new EventEmitter();
        // Stream that emits when the share popup is closed
        this.popupClosed = new EventEmitter();
    }
}
SharePopupComponent.ɵfac = function SharePopupComponent_Factory(t) { return new (t || SharePopupComponent)(ɵngcc0.ɵɵdirectiveInject(SharePopupService), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ShareService)); };
SharePopupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SharePopupComponent, selectors: [["share-popup-button"]], inputs: { theme: "theme", icon: "icon", text: "text", closeIcon: "closeIcon", showIcon: "showIcon", showText: "showText", size: "size", hasBackdrop: "hasBackdrop", include: "include", exclude: "exclude", url: "url", title: "title", description: "description", image: "image", tags: "tags", autoSetMeta: "autoSetMeta", show: "show", disabled: "disabled", backdropClass: "backdropClass" }, outputs: { opened: "opened", closed: "closed", popupClosed: "popupClosed" }, decls: 5, vars: 18, consts: [["shareButtonsPopup", "", 1, "sb-wrapper", 3, "theme", "include", "exclude", "show", "url", "showIcon", "showText", "size", "tags", "autoSetMeta", "title", "description", "image", "closed", "opened"], [1, "sb-content"], ["class", "sb-icon", 4, "ngIf"], ["class", "sb-text", 4, "ngIf"], [1, "sb-icon"], [3, "icon", "fixedWidth"], [1, "sb-text"]], template: function SharePopupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "button", 0);
        ɵngcc0.ɵɵlistener("closed", function SharePopupComponent_Template_button_closed_1_listener($event) { return ctx.closed.emit($event); })("opened", function SharePopupComponent_Template_button_opened_1_listener($event) { return ctx.opened.emit($event); });
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵtemplate(3, SharePopupComponent_div_3_Template, 2, 2, "div", 2);
        ɵngcc0.ɵɵtemplate(4, SharePopupComponent_div_4_Template, 2, 1, "div", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("sb-button sb-", ctx.theme, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("theme", ctx.theme)("include", ctx.include)("exclude", ctx.exclude)("show", ctx.show)("url", ctx.url)("showIcon", ctx.showIcon)("showText", ctx.showText)("size", ctx.size)("tags", ctx.tags)("autoSetMeta", ctx.autoSetMeta)("title", ctx.title)("description", ctx.description)("image", ctx.image);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showIcon);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showText);
    } }, directives: [ɵngcc2.SharePopupButtonsDirective, ɵngcc3.NgIf, ɵngcc4.FaIconComponent], styles: ["[_nghost-%COMP%]{--button-color:#282c34}"], changeDetection: 0 });
SharePopupComponent.ctorParameters = () => [
    { type: SharePopupService },
    { type: ShareService }
];
SharePopupComponent.propDecorators = {
    theme: [{ type: Input }],
    icon: [{ type: Input }],
    text: [{ type: Input }],
    include: [{ type: Input }],
    exclude: [{ type: Input }],
    closeIcon: [{ type: Input }],
    url: [{ type: Input }],
    title: [{ type: Input }],
    description: [{ type: Input }],
    image: [{ type: Input }],
    tags: [{ type: Input }],
    autoSetMeta: [{ type: Input }],
    show: [{ type: Input }],
    showIcon: [{ type: Input }],
    showText: [{ type: Input }],
    size: [{ type: Input }],
    disabled: [{ type: Input }],
    hasBackdrop: [{ type: Input }],
    backdropClass: [{ type: Input }],
    opened: [{ type: Output }],
    closed: [{ type: Output }],
    popupClosed: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SharePopupComponent, [{
        type: Component,
        args: [{
                selector: 'share-popup-button',
                template: `
    <div class="sb-button sb-{{theme}}">
      <button shareButtonsPopup
              class="sb-wrapper"
              [theme]="theme"
              [include]="include"
              [exclude]="exclude"
              [show]="show"
              [url]="url"
              [showIcon]="showIcon"
              [showText]="showText"
              [size]="size"
              [tags]="tags"
              [autoSetMeta]="autoSetMeta"
              [title]="title"
              [description]="description"
              [image]="image"
              (closed)="closed.emit($event)"
              (opened)="opened.emit($event)">
        <div class="sb-content">

          <div *ngIf="showIcon" class="sb-icon">
            <fa-icon [icon]="icon" [fixedWidth]="true"></fa-icon>
          </div>

          <div *ngIf="showText" class="sb-text">
            {{ text }}
          </div>

        </div>
      </button>
    </div>
  `,
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: [":host{--button-color:#282c34}"]
            }]
    }], function () { return [{ type: SharePopupService }, { type: ɵngcc1.ShareService }]; }, { theme: [{
            type: Input
        }], icon: [{
            type: Input
        }], text: [{
            type: Input
        }], closeIcon: [{
            type: Input
        }], showIcon: [{
            type: Input
        }], showText: [{
            type: Input
        }], size: [{
            type: Input
        }], hasBackdrop: [{
            type: Input
        }], opened: [{
            type: Output
        }], closed: [{
            type: Output
        }], popupClosed: [{
            type: Output
        }], include: [{
            type: Input
        }], exclude: [{
            type: Input
        }], url: [{
            type: Input
        }], title: [{
            type: Input
        }], description: [{
            type: Input
        }], image: [{
            type: Input
        }], tags: [{
            type: Input
        }], autoSetMeta: [{
            type: Input
        }], show: [{
            type: Input
        }], disabled: [{
            type: Input
        }], backdropClass: [{
            type: Input
        }] }); })();
