// ngx-light-carousel/esm2015/lib/ngx-light-carousel.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { NgxCarouselItemDirective } from './ngx-light-carousel-item.directive';
import { NgxCarouselComponent } from './ngx-light-carousel.component';
import * as ɵngcc0 from '@angular/core';
let NgxCarouselModule = class NgxCarouselModule {
};
NgxCarouselModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxCarouselModule });
NgxCarouselModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxCarouselModule_Factory(t) { return new (t || NgxCarouselModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxCarouselModule, { declarations: function () { return [NgxCarouselComponent, NgxCarouselItemDirective]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [NgxCarouselComponent, NgxCarouselItemDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxCarouselModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule],
                declarations: [NgxCarouselComponent, NgxCarouselItemDirective],
                exports: [NgxCarouselComponent, NgxCarouselItemDirective]
            }]
    }], null, null); })();
export { NgxCarouselModule };
