// paytm-blink-checkout-angular/esm5/lib/components/checkout.component.js
import { CheckoutService } from './../services/checkout.service';
import { Component } from '@angular/core';
import CONSTANTS from '../constants';
import * as ɵngcc0 from '@angular/core';
var CheckoutComponent = /** @class */ (function () {
    function CheckoutComponent(checkoutService) {
        this.checkoutService = checkoutService;
    }
    CheckoutComponent.prototype.ngAfterViewInit = function () {
        this.subs = this.checkoutService
            .checkoutJsInstance$
            .subscribe(this.invoke);
    };
    CheckoutComponent.prototype.invoke = function (checkoutJsInstance) {
        if (checkoutJsInstance && checkoutJsInstance.invoke) {
            try {
                checkoutJsInstance.invoke();
            }
            catch (error) {
                console.error(CONSTANTS.ERRORS.INVOKE, error);
            }
        }
    };
    CheckoutComponent.prototype.ngOnDestroy = function () {
        if (this.subs) {
            this.subs.unsubscribe();
        }
    };
    /** @nocollapse */
    CheckoutComponent.ctorParameters = function () { return [
        { type: CheckoutService }
    ]; };
CheckoutComponent.ɵfac = function CheckoutComponent_Factory(t) { return new (t || CheckoutComponent)(ɵngcc0.ɵɵdirectiveInject(CheckoutService)); };
CheckoutComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CheckoutComponent, selectors: [["paytm-checkout"]], decls: 2, vars: 1, consts: [[3, "id"]], template: function CheckoutComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("id", ctx.checkoutService.checkoutJsContainerId);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CheckoutComponent, [{
        type: Component,
        args: [{
                selector: 'paytm-checkout',
                template: "<div [id]=\"checkoutService.checkoutJsContainerId\"><div>"
            }]
    }], function () { return [{ type: CheckoutService }]; }, null); })();
    return CheckoutComponent;
}());
export { CheckoutComponent };
